plugin-server/src/main/pluginsServer.ts

```diff
@@ -361,38 +357,18 @@ export async function startPluginsServer(
         }
 
         if (capabilities.processAsyncWebhooksHandlers) {
-            // If we have a hub, then reuse some of it's attributes, otherwise
-            // we need to create them. We only initialize the ones we need.
-            const postgres = hub?.postgres ?? new PostgresRouter(serverConfig)
-            const kafka = hub?.kafka ?? createKafkaClient(serverConfig)
-            const teamManager = hub?.teamManager ?? new TeamManager(postgres, serverConfig)
-            const organizationManager = hub?.organizationManager ?? new OrganizationManager(postgres, teamManager)
-            const KafkaProducerWrapper = hub?.kafkaProducer ?? (await createKafkaProducerWrapper(serverConfig))
-            const rustyHook =
-                hub?.rustyHook ??
-                new RustyHook(
-                    buildIntegerMatcher(serverConfig.RUSTY_HOOK_FOR_TEAMS, true),
-                    serverConfig.RUSTY_HOOK_ROLLOUT_PERCENTAGE,
-                    serverConfig.RUSTY_HOOK_URL,
-                    serverConfig.EXTERNAL_REQUEST_TIMEOUT_MS
-                )
-            const appMetrics =
-                hub?.appMetrics ??
-                new AppMetrics(
-                    KafkaProducerWrapper,
-                    serverConfig.APP_METRICS_FLUSH_FREQUENCY_MS,
-                    serverConfig.APP_METRICS_FLUSH_MAX_QUEUE_SIZE
-                )
+            ;[hub, closeHub] = hub ? [hub, closeHub] : await createHub(serverConfig, capabilities)
```

Comment:
If you check out the blame you can see https://github.com/PostHog/posthog/commit/d77858d275e0cc03ac5ea429eb81334cab7af22e we're wanting to rather move away from hub than add more stuff to it.

Reply:
Hmmm. Not meaning to get into a big refactor but the "hub" is essentially your dependency injection. Getting around it seems to be making the code pretty hard to parse and manage 😅 Do you have an alternative solution? Or is it just the nasty passing of variables everywhere?

Reply:
looking at this again - not sure I follow on the big refactor part, we'll want to not initialise the action matcher in the hub, so moving it here and starting it here would be easy, no?

Reply:
so in this code path we're just creating a hub to throw it away immediately? 
Instead, I'd move this here after PG creation, or am I missing something?
```
    // we need to create them. We only initialize the ones we need.
    const postgres = hub?.postgres ?? new PostgresRouter(serverConfig)
    const actionManager = new ActionManager(postgres)
    await actionManager.prepare()
    const actionMatcher = new ActionMatcher(postgres, actionManager)
```

Reply:
So in that area the code makes no sense to me at all 😅 The `hub` is basically never there and if it isn't we have all the duplocate setup code. For reasons that are super unclear as the hub is mostly just instantiating things (like a DI)

Reply:
If the plugin-server being set-up has ingestion capabilities (e.g. local dev, hobby deploy), then hub is created in an earlier if statement. 

We want to move away from the hub and have more granular dependency injection, this is what Harry's PR was in direction for. We're approaching this via "leave the campsite cleaner than you found it".